qgis/examples/elevation_tile_downloader/gui/tile_downloader_dlg.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+ElevationTileDownloaderDialog
+A QGIS plugin
+Downloads elevation tiles
+Generated by Plugin Builder: http://g-sherman.github.io/Qgis-Plugin-Builder/
+                             -------------------
+        begin                : 2021-09-01
+        git sha              : $Format:%H$
+        copyright            : (C) 2021 by Christian Beiwinkel
+        email                : christian@gis-ops.com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from pathlib import Path
+
+from qgis.PyQt.QtCore import Qt
+from qgis.PyQt.QtWidgets import QDialogButtonBox, QProgressBar, QDialog
+from qgis.gui import QgisInterface, QgsMessageBarItem
+from qgis.core import (
+    QgsProject,
+    QgsVectorLayer,
+    QgsGeometry,
+    QgsMapLayerProxyModel,
+    QgsApplication,
+    QgsRectangle,
+    QgsCoordinateTransform,
+    QgsCoordinateReferenceSystem,
+    Qgis,
+    QgsTask,
+)
+from math import floor, ceil
+from typing import List, Optional
+
+from ..gui.tile_downloader_dlg_base import Ui_ElevationTileDownloaderDialogBase
+from ..core.task import DownloadTask
+
+
+def to_wgs84(
+    geometry: QgsGeometry, own_crs: QgsCoordinateReferenceSystem
+) -> QgsGeometry:
+    """
+    Transforms the ``point`` to (``direction=ForwardTransform``) or from
+    (``direction=ReverseTransform``) WGS84.
+    """
+    wgs84 = QgsCoordinateReferenceSystem.fromEpsgId(4326)
+    if own_crs != wgs84:
+        xform = QgsCoordinateTransform(own_crs, wgs84, QgsProject.instance())
+        geometry.transform(xform)
+    return geometry
+
+
+def create_grid_from_bounds(bounds: QgsRectangle) -> List[QgsGeometry]:
+    """Creates a regular grid of size 1x1 within specified bounds"""
+    grid_bboxes = []
+
+    # loop through x and y range and create the grid
+    min_x, min_y = [floor(x) for x in (bounds.xMinimum(), bounds.yMinimum())]
```

Comment:
I very much liked the `floor` & `ceil` uses here! I just re-arranged them away from the main method